packages/js-drive/lib/masternodes/manageMasternodesIdentities.js

```diff
@@ -0,0 +1,205 @@
+
+        if (proRegTxInfo.operatorReward > 0) {
+          const operatorIdentityId = hash(Buffer.concat(
+            Buffer.from(masternodeEntry.proRegTxHash, 'hex'),
+            Buffer.from(proRegTxInfo.state.pubKeyOperator, 'hex'),
+          ));
+
+          const operatorIdentity = await stateRepository.fetchIdentity(operatorIdentityId);
+
+          //  Create an identity for operator if there is no identity exist with the same ID
+          if (operatorIdentity === null) {
+            await createIdentity(
+              operatorIdentityId,
+              Buffer.from(proRegTxInfo.state.pubKeyOperator, 'hex'),
+              IdentityPublicKey.TYPES.BLS12_381,
+            );
+          }
+
+          // Delete document from rewards data contract with ID corresponding to the
+          // masternode identity (ownerId) and previous operator identity (payToId)
+
+          const previousOperatorData = previousMNList
+            // eslint-disable-next-line max-len
+            .find((previousMnListEntry) => previousMnListEntry.proRegTxHash === masternodeEntry.proRegTxHash);
+
+          const previousOperatorIdentityId = hash(Buffer.concat(
+            Buffer.from(previousOperatorData.proRegTxHash, 'hex'),
+            Buffer.from(proRegTxInfo.state.pubKeyOperator, 'hex'),
+          ));
+
+          documentsToDelete.push(transactionalDpp.document.create(
+            contract,
+            Buffer.from(masternodeEntry.proRegTxHash, 'hex'),
+            'masternodeRewardShares',
+            {
+              payToId: previousOperatorIdentityId,
+              percentage: proRegTxInfo.operatorReward,
+            },
+          ));
+
+          // Create a document in rewards data contract with percentage defined
+          // in corresponding ProRegTx
+          documentsToCreate.push(transactionalDpp.document.create(
+            contract,
+            Buffer.from(masternodeEntry.proRegTxHash, 'hex'),
+            'masternodeRewardShares',
+            {
+              payToId: operatorIdentityId,
+              percentage: proRegTxInfo.operatorReward,
+            },
+          ));
+        }
+      }),
+    );
+
+    // A masternode disappeared or is not valid
+    const disappearedOrInvalidMasterNodes = previousMNList
+      .filter((previousMnListEntry) => !currentMNList
+        // eslint-disable-next-line max-len
+        .find((currentMnListEntry) => currentMnListEntry.proRegTxHash === previousMnListEntry.proRegTxHash))
+      .concat(currentMNList.filter((currentMnListEntry) => !currentMnListEntry.isValid));
+
+    await Promise.all(
+      disappearedOrInvalidMasterNodes.map(async (masternodeEntry) => {
+        const rawTransaction = await stateRepository
+          .fetchTransaction(masternodeEntry.proRegTxHash);
+
+        const { extraPayload: proRegTxInfo } = new Transaction(rawTransaction.data);
+
+        if (proRegTxInfo.operatorReward > 0) {
```

Comment:
You need to delete all documents belonging to masternode identity even if proRegTxInfo.operatorReward == 0 because masternode can manually create documents in this contract and share rewards with any identity.